### ISSUE TITLE
TypeHintDeclarationSniff settings

### ISSUE THREAD

**Issue by lookyman** [slevomat/coding-standard#123]
I am on PHP 7.1, but I put this in my `ruleset.xml` for testing purposes:

```xml
<rule ref="SlevomatCodingStandard.TypeHints.TypeHintDeclaration">
    <properties>
        <property name="enableNullableTypeHints" value="false"/>
        <property name="enableVoidTypeHint" value="false"/>
    </properties>
</rule>
```

This however doesn't seem to have any effect, even though I have nullable types and void returns in my codebase. Shouldn't it scream at me, or is there something I am missing?

**Comment by ondrejmirtes** [comment]
Please post a code example you expect it to report.

This sniff converts for example string|null phpDoc into ?string native
typehint when enabled.

**Comment by kukulich** [comment]
These properties should help you to migrate code from 7.0 to 7.1. It doesn't make any sense for me to migrate from 7.1. to 7.0, so if you turn off these properties, it doesn't report anything :)

**Comment by lookyman** [comment]
Yep, I thought those properties forbid nullable types and void returns if they are turned off, but they just force them when they are on. My bad, thanks!

**Comment by ondrejmirtes** [comment]
There's no reason to forbid nullable typehints and voids.
